rdk/rdk.py

```diff
@@ -1259,6 +1280,109 @@ def deploy(self):
 
         return 0
 
+    def export(self):
+
+        self.__parse_export_args()
+
+        # get the rule names
+        rule_names = self.__get_rule_list_for_command("export")
+
+        # run the export code
+        print("Running export")
+
+        for rule_name in rule_names:
+            rule_params, cfn_tags = self.__get_rule_parameters(rule_name)
+
+            if 'SourceIdentifier' in rule_params:
+                print("Found Managed Rule, Ignored.")
+                print("Export support only Custom Rules.")
+                continue
+
+            source_events = []
+            if 'SourceEvents' in rule_params:
+                source_events = [rule_params['SourceEvents']]
+
+            source_periodic = "NONE"
+            if 'SourcePeriodic' in rule_params:
+                source_periodic = rule_params['SourcePeriodic']
+
+            combined_input_parameters = {}
+            if 'InputParameters' in rule_params:
+                combined_input_parameters.update(json.loads(rule_params['InputParameters']))
+
+            if 'OptionalParameters' in rule_params:
+                # Remove empty parameters
+                keys_to_delete = []
+                optional_parameters_json = json.loads(rule_params['OptionalParameters'])
+                for key, value in optional_parameters_json.items():
+                    if not value:
+                        keys_to_delete.append(key)
+                for key in keys_to_delete:
+                    del optional_parameters_json[key]
+                combined_input_parameters.update(optional_parameters_json)
+
+            print("Found Custom Rule.")
+            s3_src = ""
+            s3_dst = self.__package_function_code(rule_name, rule_params)
+
+            layers = []
+            rdk_lib_version = "0"
+            if 'SourceRuntime' in rule_params:
+                if rule_params['SourceRuntime'] == "python3.6-lib":
+                    if self.args.rdklib_layer_arn:
+                        layers.append(self.args.rdklib_layer_arn)
+                    else:
+                        rdk_lib_version = RDKLIB_LAYER_VERSION[my_session.region_name]
```

Comment:
NameError: name 'my_session' is not defined

Reply:
> Thanks for contribution!! Please also add documentation in /doc/reference for new commands

Updated the documentation.

Reply:
> NameError: name 'my_session' is not defined

Resolved this error and updated the PR. Thanks for the quick reply.